File: ios/TravelTranslator/Models/Phrase.swift
import Foundation

/// 场景短语。
struct Phrase: Identifiable, Codable, Hashable {
    let id: String
    let sceneCategory: String
    let subcategory: String?
    let sourceText: String
    let targetText: String
    let sourceLanguage: String
    let targetLanguage: String
    let transliteration: String?
    let isCustom: Bool
    let priority: Int

    enum CodingKeys: String, CodingKey {
        case id
        case sceneCategory = "scene_category"
        case subcategory
        case sourceText = "source_text"
        case targetText = "target_text"
        case sourceLanguage = "source_language"
        case targetLanguage = "target_language"
        case transliteration
        case isCustom = "is_custom"
        case priority
    }
}

struct PhrasePackage: Codable {
    let language: String
    let total: Int
    let phrases: [Phrase]
}

File: backend/app/schemas/phrase.py
"""场景短语相关 schema。"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PhraseBase(BaseModel):
    """短语基础字段。"""

    scene_category: str = Field(..., description="场景分类：restaurant/transport/hotel 等")
    subcategory: str | None = None
    source_text: str
    target_text: str
    source_language: str = "zh"
    target_language: str
    transliteration: str | None = None


class PhraseCreate(PhraseBase):
    """创建自定义短语。"""

    user_id: str | None = None
    is_custom: bool = True


class PhraseOut(PhraseBase):
    """短语输出。"""

    model_config = ConfigDict(from_attributes=True)

    id: str
    is_custom: bool
    use_count: int
    priority: int
    created_at: datetime


class PhrasePackage(BaseModel):
    """场景短语包（按语言打包）。"""

    language: str
    destination: str | None = None
    version: str = "1.0.0"
    total: int
    phrases: list[PhraseOut]
